File: src/main/java/xyz/deseteral/lebkuchenfm/api/commands/text/TextCommandResponseDtoMapper.java
package xyz.deseteral.lebkuchenfm.api.commands.text;

import xyz.deseteral.lebkuchenfm.domain.CommandProcessingResponse;
import xyz.deseteral.lebkuchenfm.domain.TextIsNotACommandException;
import xyz.deseteral.lebkuchenfm.domain.NoSuchCommandProcessorException;

final class TextCommandResponseDtoMapper {
    public static TextCommandResponseDto from(CommandProcessingResponse processingResponse) {
        return new TextCommandResponseDto(processingResponse.getResponse());
    }

    public static TextCommandResponseDto from(TextIsNotACommandException exception) {
        return new TextCommandResponseDto(exception.getMessage());
    }

    public static TextCommandResponseDto from(NoSuchCommandProcessorException exception) {
        return new TextCommandResponseDto(exception.getMessage());
    }
}

File: src/main/kotlin/xyz/deseteral/lebkuchenfm/api/commands/text/TextCommandResponseDtoMapper.kt
package xyz.deseteral.lebkuchenfm.api.commands.text

import xyz.deseteral.lebkuchenfm.domain.CommandProcessingResponse
import xyz.deseteral.lebkuchenfm.domain.TextIsNotACommandException
import xyz.deseteral.lebkuchenfm.domain.NoSuchCommandProcessorException

internal object TextCommandResponseDtoMapper {
    fun from(processingResponse: CommandProcessingResponse): TextCommandResponseDto {
        return TextCommandResponseDto(processingResponse.response)
    }

    fun from(exception: TextIsNotACommandException): TextCommandResponseDto {
        return TextCommandResponseDto(exception.message)
    }

    fun from(exception: NoSuchCommandProcessorException): TextCommandResponseDto {
        return TextCommandResponseDto(exception.message)
    }
}
